### PR TITLE
Fix pie chart in demo_agg_filter.py

### PR DESCRIPTION
**Pull request by leejjoon**
## PR Summary
This fixes the pie chart in [demo_agg_filter.py] (https://matplotlib.org/stable/gallery/misc/demo_agg_filter.html) which is incorrectly displayed with simple black color . Fixed image is shown below. It also add new `shrink` parameter and a bit of more documentation.

image

## PR Checklist

**Tests and Styling**
- [N/A] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [N/A] New features are documented, with examples if plot related.
- [N/A] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [N/A] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [N/A] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

**Review by oscargus — APPROVED** [review]

**Comment by ksunden** [comment]
The implementation of `shrink` relies on `scipy`, which I do not believe we want to make a dependency for building the docs.

See, for example, the [double pendulum example](https://matplotlib.org/stable/gallery/animation/double_pendulum.html) which implements the code in python rather than using a scipy method which would provide better results.

**Comment by oscargus** [comment]
Good observation! (Should probably not approve stuff with a heavy cold...)

It would be great to get rid of the scipy dependency.

**Comment by leejjoon** [comment]
> The implementation of `shrink` relies on `scipy`, which I do not believe we want to make a dependency for building the docs.

Would it be okay if the example uses "shrink=None" so that importing scipy is bypassed?
I think the `shrink` option gives a more pleasing result, and it would be good if we can let the users play with it.
If this is not acceptable, I can simply remove the shrink option and its dependency on scipy.

**Comment by tacaswell** [comment]
Welcome back @leejjoon !

**Comment by ksunden** [comment]
I think the effect you are looking for can be had with library-native changes to the `explode` parameter rather than passing through `scipy.ndimage`.

Figure_1

Here is a quick example demonstrating the differences between the example as is (with your fixes so that it works, but without `shrink`, with your shrink implementation and with no `shrink`, but a uniform `0.1` for `explode` rather than `(0, 0.05, 0, 0)`

**Comment by leejjoon** [comment]
Okay, I have removed shrink parameter so the scipy dependency. However, I still think that it would be good to have a `shrink` parameter as it offers a rather general solution not specific to pie chart.

I also have rebased the PR to the current main and force-pushed it.

**Comment by leejjoon** [comment]
> Welcome back @leejjoon !

Thanks!

File: examples/misc/demo_agg_filter.py [hunk 1]
```diff
@@ -99,8 +99,19 @@ def process_image(self, padded_src, dpi):
...
-    def __init__(self, sigma, fraction=0.5):
+    """Apply LightSource filter"""
+
+    def __init__(self, sigma, fraction=1):
+        """
+        Parameters
+        ----------
+        sigma : float
+            sigma for gaussian filter
+        fraction: number: deault 1
```
Comment (QuLogic) [inline examples/misc/demo_agg_filter.py:110]:
```suggestion
        fraction: number, default: 1
```

**Review by QuLogic — APPROVED** [review]
LGTM, https://output.circle-artifacts.com/output/job/cad9b940-87df-45de-9e64-3f9ddf95ffd1/artifacts/0/doc/build/html/gallery/misc/demo_agg_filter.html

**Pull request merged**